### ISSUE TITLE
[Question] Reconstructed pixels are discontinuous at patch boundaries.

### ISSUE THREAD

**Issue by LeroyChou** [facebookresearch/mae#153]
Hi, thank you for this great work! 

I test the code on my custom dataset,  and find that the pixels on reconstructed image (before pasting visbile patches on it) is not continuous on the patch boundaries. So it looks like there are many (e.g. 14x14=196) grids on the image. May I know how to solve it? 

Kind regards.

**Comment by stha-prashant** [comment]
@LeroyChou, I'm facing the same problem on my custom dataset, did you figure out how to solve it?

**Comment by daisukelab** [comment]
@LeroyChou @stha-prashant Are you pre-training with `--norm_pix_loss` option? If so, try without using it.

**Comment by LeroyChou** [comment]
> @LeroyChou, I'm facing the same problem on my custom dataset, did you figure out how to solve it?

Not really. I tried several methods, but they look useless. But if you spend more time on training, the reconstructed images will be better.

**Comment by LeroyChou** [comment]
> @LeroyChou @stha-prashant Are you pre-training with `--norm_pix_loss` option? If so, try without using it.

Thank you for replying. But I didn't train with `--norm_pix_loss`.

**Comment by daisukelab** [comment]
@LeroyChou Then there might be some different causes.

FYI, I'd share my results. I guess yours might look different from ours.
We have confirmed MAE reconstruction using an audio spectrogram in two of our repos; one doesn't use the option, and another uses it.

When not using the norm_pix_loss, we see the reconstruction clean. --> https://github.com/nttcslab/msm-mae/blob/main/misc/Note_visualization.ipynb
<img width="880" alt="image" src="https://github.com/facebookresearch/mae/assets/14831220/4c7ccee8-fed1-4a08-bbad-ada82674e896">
(The upper row is the input, lower is the reconstruction. The white square is a visible patch, and non-white square patches are the reconstructed ones.)

When using the norm_pix_loss, we see patches are discontinuous at their edges; simply because MAE has learned from normalized patches. Please note that each patch is normalized independently, simply resulting in discontinuous edges. --> https://github.com/nttcslab/m2d/blob/master/Note_viz_msm_mae.ipynb
<img width="892" alt="image" src="https://github.com/facebookresearch/mae/assets/14831220/fe320cc7-76b3-43e7-8127-be8a44268431">

**Comment by LeroyChou** [comment]
@daisukelab Your work appears to be quite intriguing!

In fact, our situation differs slightly from yours. Initially, I trained the model using the `--norm_pix_loss option`, but the reconstructed image was unsatisfactory --  it was filled with colored noise. I did not save the image, but it resembled this: 
![image](https://github.com/facebookresearch/mae/assets/34728480/5d9bfb21-fb94-4348-8fe1-6a9087d1057c)
The second time, I trained without the` --norm_pix_loss option`. The colored noise disappeared, but discontinuity on the patch boundaries became apparent.

I am not well-versed in the field of audio spectrograms. May I inquire if you have ever encountered colored noise? What is the range of audio spectrogram data?

**Comment by daisukelab** [comment]
@LeroyChou No, I have not experienced the colored noise issue, at least with a model finished pre-training. I also have tested on ImageNet (with norm_pix_loss) without having related issues.
My spectrograms are standardized to become N(0, 1), mostly in the [-1,1] range.

I guess I remember (vaguely) that the decoder in the very early stage might output a noise-like image.

**Comment by LeroyChou** [comment]
@daisukelab Thank you for your response. You are correct that the early stage generates noisy images. And as training progresses, the noise disappears. It's quite strange that we are experiencing different situations despite using similar models.I have also noticed that others have encountered same problem with colored noises, as mentioned in this issue: https://github.com/facebookresearch/mae/issues/95. One possibility is that there may be significant differences between our raw data, specifically between audio spectrograms and RGB images. Do you have any thoughts?

**Comment by daisukelab** [comment]
@LeroyChou I think of nothing, while one apparent difference might be that we handle single-channel data instead of RGB channels in the image.
BTW, I found an old reconstruction log when the model was trained after one epoch out of 300.
It's a blurry image instead of noises. Let me fix my comment above, and what I have experienced is that MAE has outputted blurry images in the early training epochs instead of noisy ones.
![image](https://github.com/facebookresearch/mae/assets/14831220/0564336a-605f-42c4-856d-1075a5ef3c9d)
(yellows show that they are reconstructed patches.)
